src/backend/distributed/commands/multi_copy.c

```diff
@@ -1295,6 +1661,49 @@ FinalizeCopyToNewShard(ShardConnections *shardConnections)
 }
 
 
+/*
+ * MasterUpdateShardStatistics dispatches the update shard statistics call
+ * between local or remote master node according to the master connection state.
+ */
+static void
+MasterUpdateShardStatistics(uint64 shardId)
+{
+	if (masterConnection == NULL)
+	{
+		UpdateShardStatistics(shardId);
+	}
+	else
+	{
+		RemoteUpdateShardStatistics(shardId);
+	}
+}
+
+
+/*
+ * RemoteUpdateShardStatistics updates shard statistics on the remote master node.
+ */
+static void
+RemoteUpdateShardStatistics(uint64 shardId)
+{
+	PGresult *queryResult = NULL;
+
+	StringInfo updateShardStatisticsCommand = makeStringInfo();
+	appendStringInfo(updateShardStatisticsCommand, UPDATE_SHARD_STATISTICS_QUERY,
+					 shardId);
+
+	queryResult = PQexec(masterConnection, updateShardStatisticsCommand->data);
+	if (PQresultStatus(queryResult) != PGRES_TUPLES_OK)
+	{
+		ereport(WARNING, (errmsg("couldn't update shard statistics"),
```

Comment:
This rolls back the transaction that is active on masterConnection. We'd better just error out here, because we don't want to report the COPY succeeded when  it didn't.